sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java

```diff
@@ -338,17 +368,20 @@ public TableClientBuilder credential(AzureNamedKeyCredential credential) {
     }
 
     /**
-     * Sets the {@link TokenCredential} used to authorize requests sent to the service. Setting this is mutually
-     * exclusive with using {@link TableClientBuilder#credential(AzureNamedKeyCredential)},
-     * {@link TableClientBuilder#credential(AzureSasCredential)} or
-     * {@link TableClientBuilder#sasToken(String)}.
+     * Sets the {@link TokenCredential} used to authorize requests sent to the service. Refer to the Azure SDK for Java
+     * <a href="https://aka.ms/azsdk/java/docs/identity">identity and authentication</a>
+     * documentation for more details on proper usage of the {@link TokenCredential} type.
+     *
+     * Setting this is mutually exclusive with using {@link #credential(AzureNamedKeyCredential)},
+     * {@link #credential(AzureSasCredential)} or {@link #sasToken(String)}.
      *
      * @param credential {@link TokenCredential} used to authorize requests sent to the service.
      *
      * @return The updated {@link TableClientBuilder}.
      *
      * @throws NullPointerException If {@code credential} is {@code null}.
      */
+    @Override
```

Comment:
Setting a custom `HttpPipeline` would make it so the builder ignores the value set in this method. We should probably add a note like you did in other methods on this same class.